DllImportGenerator/DllImportGenerator/Marshalling/NonBlittableArrayMarshaller.cs

```diff
@@ -135,13 +145,46 @@ public override IEnumerable<StatementSyntax> Generate(TypePositionInfo info, Stu
                                         ArgumentList())),
                             MarshallerHelpers.GetForLoop(managedLocal, IndexerIdentifier)
                                 .WithStatement(Block(
-                                    List(_elementMarshaller.Generate(info with { ManagedType = GetElementTypeSymbol(info) }, arraySubContext))))));
+                                    List(_elementMarshaller.Generate(
+                                        info with { ManagedType = GetElementTypeSymbol(info), RefKind = elementRefKind },
+                                        arraySubContext))))));
                     }
                     break;
                 case StubCodeContext.Stage.Unmarshal:
-                    if (info.IsManagedReturnPosition || (info.IsByRef && info.RefKind != RefKind.In))
+                    if (info.IsManagedReturnPosition
+                        || (info.IsByRef && info.RefKind != RefKind.In)
+                        || (info.ByValueContentsMarshalKind & ByValueContentsMarshalKind.Out) != 0)
```

Comment:
Can't we use the `HasFlag()` function for all of these checks with `ByValueContentsMarshalKind`?